### PR TITLE
update power points during creation

### PR DESCRIPTION
**Pull request by andrewljohnson**
This makes it so the "power points" of a card updates when you edit the cost or power/toughness in the card builder. FYI, a card can have 0-100 power points, and a deck can have cards worth up to 100 power points in it.

Most of that involved refactoring the cardInfo method.

### CODE REVIEW

File: create_cards/views.py [hunk 1]
```diff
@@ -234,7 +253,7 @@ def create_card_save(request, required_key, event_name):
                 card_info["cost"] = int(card_info["cost"])    
             if "effects" in card_info:
                 for e in card_info["effects"]:
-                    if "amount" in e and e["amount"] != None:
+                    if e.get("amount") is not None:
```
Comment (andrewljohnson) [inline create_cards/views.py:256]: I failed to check in this change last PR.

File: static/js/components/GameRoom.js [hunk 2]
```diff
@@ -12,10 +12,10 @@ export class GameRoom {
     }
 
     connect() {
-        if (this.gameSocket == null) {
+        if (this.gameSocket === null) {
```
Comment (andrewljohnson) [inline static/js/components/GameRoom.js:15]: I failed to check in this change last PR.

File: static/js/components/GameRoom.js [hunk 3]
```diff
@@ -12,10 +12,10 @@ export class GameRoom {
...
             this.setupSocket();
         }
-        if (this.gameSocket.readyState == WebSocket.OPEN) {
+        if (this.gameSocket.readyState === WebSocket.OPEN) {
```
Comment (andrewljohnson) [inline static/js/components/GameRoom.js:18]: I failed to check in this change last PR.

File: static/js/components/GameRoom.js (hunk 2 above)
Comment (andrewljohnson) [inline static/js/components/GameRoom.js:15]: I failed to check in the changes to this file last PR. These === are unrelated

File: create_cards/views.py [hunk 4]
```diff
@@ -172,6 +172,25 @@ def get_effect_for_info(request):
...
+            card_info["cost"] = int(card_info["cost"])
+        if "strength" in card_info:
+            card_info["strength"] = int(card_info["strength"])
+        if "hit_points" in card_info:
+            card_info["hit_points"] = int(card_info["hit_points"])
+        Analytics.log_amplitude(request, "Create Card Get Power Points", {})
+        power_points = Card(card_info).power_points_value()
+        return JsonResponse({"power_points": power_points})
+    else:
+        return JsonResponse({"error": "Unsupported request type"})
```
Comment (mcglincy) [inline create_cards/views.py:192]: I haven't looked how everything is wired up yet, but usually "what http methods do I support" is handled by implementing get() vs post(), or annotations. You also just check if != post and return at the beginning of the method and then do the rest.

File: static/js/views/create_cards/CardBuilderBase.js [hunk 5]
```diff
@@ -130,10 +130,42 @@ export class CardBuilderBase {
         return description;
     }
 
+    getPowerPoints() {
+        Constants.postData(`${this.baseURL()}/get_power_points`, { card_info: this.cardInfo(), card_id: this.cardID })
```
Comment (mcglincy) [inline static/js/views/create_cards/CardBuilderBase.js:134]: one for the TODO list: use async/await in Javascript. Doesn't have to be now tho.

File: static/js/views/create_cards/CardBuilderCost.js [hunk 6]
```diff
@@ -14,18 +14,9 @@ export class CardBuilderCost extends CardBuilderBase {
...
-            image: this.originalCardInfo.image, 
-            effects: this.originalCardInfo.effects, 
-            strength: this.originalCardInfo.strength, 
-            hit_points: this.originalCardInfo.hit_points, 
-            description: this.cardDescription(),
-            cost: this.cardCost(), 
-            power_points: this.powerPoints ? this.powerPoints : this.originalCardInfo.power_points,
-            author_username: this.originalCardInfo.author_username
-        };
+        let info = super.cardInfo();
```
Comment (mcglincy) [inline static/js/views/create_cards/CardBuilderCost.js:17]: const

**Review by mcglincy — APPROVED** [review]
Seems OK; I don't yet have enough context to make more meaningful comments.

File: create_cards/views.py (hunk 4 above)
Comment (andrewljohnson) [reply create_cards/views.py:192]: good thought, i switched this views file to use request_POST decorator throughout

File: static/js/views/create_cards/CardBuilderBase.js (hunk 5 above)
Comment (andrewljohnson) [reply static/js/views/create_cards/CardBuilderBase.js:134]: This is using async, but it uses it with then() - why is async/await better?

File: static/js/views/create_cards/CardBuilderCost.js (hunk 6 above)
Comment (andrewljohnson) [reply static/js/views/create_cards/CardBuilderCost.js:17]: done, though meandering thought is const not protecting object keys and array values is weird, dubious, and or smelly

**Pull request merged**